tests/test_modeling_blenderbot.py

```diff
@@ -164,14 +157,153 @@ def get_config_data(self):
+                                            config.vocab_size, learn_positional_embeddings=True, variant='prelayernorm',n_positions=config.max_position_embeddings,
+                                            activation=config.activation_function, dropout=config.dropout, embedding=embeddings, reduction_type=None)
+        parlai_encoder.eval()
+        
+        self._copy_layer_weights_in_blender_encoder(blender_encoder, parlai_encoder, config.encoder_layers)
+        
+        expected_output = parlai_encoder(input_ids)[0]
+        blender_output = blender_encoder(input_ids, attention_mask=mask)[0]
+        self.assertTrue(torch.allclose(expected_output, blender_output, atol=1e-3))
+        
+    def test_blenderbot_decoder_layer_forward(self):
+        config, input_ids, mask = self.get_config_and_data()
+        
+        torch.manual_seed(0)
+        embeddings = torch.nn.Embedding(config.vocab_size, config.d_model, padding_idx=config.pad_token_id)
+        
+        blenderbot_model = BlenderbotForConditionalGeneration(config).to(torch_device)
+        blender_decoder_layer = blenderbot_model.decoder.layers[0]
+        blender_decoder_layer.eval()
+        
+        parlai_decoder_layer = TransformerDecoderLayer(config.encoder_attention_heads, config.d_model, config.encoder_ffn_dim, 
+                                                       dropout=config.dropout, activation='gelu', variant='prelayernorm')
+        parlai_decoder_layer.eval()
+        
+        blender_encoder = blenderbot_model.encoder
+        #blender_encoder.embed_tokens = embeddings
+        blender_encoder.eval()
+        
+        parlai_encoder = TransformerEncoder(config.encoder_attention_heads, config.encoder_layers, config.d_model, config.encoder_ffn_dim,
+                                            config.vocab_size, learn_positional_embeddings=True, variant='prelayernorm',n_positions=config.max_position_embeddings,
+                                            activation=config.activation_function, dropout=config.dropout, embedding=embeddings, reduction_type=None)
+        parlai_encoder.eval()
+        
+        self._copy_layer_weights_in_blender_encoder(blender_encoder, parlai_encoder, config.encoder_layers)
+        
+        expected_encoder_output = parlai_encoder(input_ids)[0]
+        blender_encoder_output = blender_encoder(input_ids, attention_mask=mask)[0]
+        self.assertTrue(torch.allclose(expected_encoder_output, blender_encoder_output, atol=1e-3))
+        
+        self._copy_layer_weights_in_blender_decoder_layer(blender_decoder_layer, parlai_decoder_layer)
+        tensor = embeddings(input_ids)
+        expected_decoder_layer_output = parlai_decoder_layer(tensor, encoder_output=expected_encoder_output, encoder_mask=mask)
+        
+        blender_decoder_layer_output = blender_decoder_layer(tensor.transpose(1,0), encoder_hidden_states=blender_encoder_output, encoder_attn_mask=mask)[0] 
```

Comment:
this returns `nan` values and that is due to the fact that [this](https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_bart.py#L398) returns an `-inf` tensor